# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderScrollData.h
# Commit: 77ef8e8bc669
# Full Hash: 77ef8e8bc669c9ecab4d94505b11b3cb732446c5
# Author: Alexis Beingessner <a.beingessner@gmail.com>
# Date: 2019-07-15 21:39:46
# Regressor Bug: 1547351
# File Overlap Count: 2
# Description:
#   Bug 1547351 - Automatically set content process render root r=kats
#   
#   This implements the idea of automatically setting a content proc's
#   render root based on the render root enclosing the iframe that
#   points to it. There was a bit of cleanup in here that was a bit
# ==============================================================================

diff -r b594f8e47025 -r 77ef8e8bc669 gfx/layers/wr/WebRenderScrollData.h
--- a/gfx/layers/wr/WebRenderScrollData.h	Mon Jul 15 13:20:31 2019 +0000
+++ b/gfx/layers/wr/WebRenderScrollData.h	Mon Jul 15 13:18:44 2019 +0000
@@ -21,6 +21,7 @@
 #include "mozilla/layers/RenderRootBoundary.h"
 #include "mozilla/layers/WebRenderMessageUtils.h"
 #include "mozilla/webrender/WebRenderTypes.h"
+#include "mozilla/HashTable.h"
 #include "mozilla/Maybe.h"
 #include "nsTArrayForwardDeclare.h"
 
@@ -223,7 +224,7 @@
   // valid on both the child and parent.
   // The key into this map is the scrollId of a ScrollMetadata, and the value is
   // an index into the mScrollMetadatas array.
-  std::map<ScrollableLayerGuid::ViewID, size_t> mScrollIdMap;
+  HashMap<ScrollableLayerGuid::ViewID, size_t> mScrollIdMap;
 
   // A list of all the unique ScrollMetadata objects from the layer tree. Each
   // ScrollMetadata in this list must have a unique scroll id.